superset-frontend/src/explore/components/controls/OptionControls/index.tsx

```diff
@@ -103,21 +103,86 @@ export const LabelsContainer = styled.div`
   border-radius: ${({ theme }) => theme.gridUnit}px;
 `;
 
+const borderPulse = keyframes`
+  0% {
+    right: 100%;
+  }
+  50% {
+    left: 4px;
+  }
+  90% {
+    right: 4px;
+  }
+  100% {
+    left: 100%;
+  }
+`;
+
 export const DndLabelsContainer = styled.div<{
   canDrop?: boolean;
   isOver?: boolean;
+  isDragging?: boolean;
+  isLoading?: boolean;
 }>`
+  position: relative;
   padding: ${({ theme }) => theme.gridUnit}px;
-  border: ${({ canDrop, isOver, theme }) => {
-    if (canDrop) {
-      return `dashed 1px ${theme.colors.info.dark1}`;
+  border: ${({ canDrop, isDragging, isLoading, theme }) => {
+    if (!isLoading && isDragging) {
+      return `dashed 1px ${
+        canDrop ? theme.colors.info.dark1 : theme.colors.error.dark1
+      }`;
     }
-    if (isOver && !canDrop) {
-      return `dashed 1px ${theme.colors.error.dark1}`;
-    }
-    return `solid 1px ${theme.colors.grayscale.light2}`;
+    return `solid 1px ${
+      isLoading && isDragging
+        ? theme.colors.warning.light1
+        : theme.colors.grayscale.light2
+    }`;
   }};
   border-radius: ${({ theme }) => theme.gridUnit}px;
+  &:before,
+  &:after {
+    content: ' ';
+    position: absolute;
+    border-radius: ${({ theme }) => theme.gridUnit}px;
+  }
+  &:before {
+    display: ${({ isDragging, isLoading }) =>
+      isDragging || isLoading ? 'block' : 'none'};
+    background-color: ${({ theme, canDrop }) =>
+      canDrop ? theme.colors.primary.base : theme.colors.error.light1};
+    z-index: ${({ theme }) => theme.zIndex.aboveDashboardCharts};
+    opacity: ${({ theme }) => theme.opacity.light};
+    top: 1px;
+    right: 1px;
+    bottom: 1px;
+    left: 1px;
+
+    ${({ theme, isLoading }) =>
+      isLoading &&
+      css`
+        animation: ${borderPulse} 2s ease-in infinite;
+        background: linear-gradient(currentColor 0 0) 0 100%/0% 3px no-repeat;
+        background-size: 100% ${theme.gridUnit / 2}px;
+        top: auto;
+        right: ${theme.gridUnit}px;
+        left: ${theme.gridUnit}px;
+        bottom: -${theme.gridUnit / 2}px;
+        height: ${theme.gridUnit / 2}px;
+      `}
+  }
+  &:after {
+    display: ${({ isOver, canDrop, isLoading }) =>
+      isLoading || (canDrop && isOver) ? 'block' : 'none'};
+    background-color: ${({ theme, isLoading }) =>
+      isLoading ? theme.colors.grayscale.light3 : theme.colors.primary.base};
+    z-index: ${({ theme }) => theme.zIndex.dropdown};
```

Comment:
Could you move the variables to the top? Like:

```
const TooltipSectionWrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.typography.sizes.s}px;

    &:not(:last-of-type) {
      margin-bottom: ${theme.gridUnit * 2}px;
    }
  `}
`;
```

to avoid repeating:

```
${({ theme, isDragging, isLoading }) => ... };
```